packages/ui/src/IconButton.tsx

```diff
@@ -1,183 +1,155 @@
-import React, {forwardRef, useState} from "react";
-import {Platform, Pressable, View, ViewStyle} from "react-native";
+import FontAwesome6 from "@expo/vector-icons/FontAwesome6";
+import debounce from "lodash/debounce";
+import React, {FC, useContext, useState} from "react";
+import {ActivityIndicator, Pressable, View} from "react-native";
 
-import {Box} from "./Box";
-import {IconButtonProps, IconColor, iconSizeToNumber, SurfaceColor} from "./Common";
-import {Icon} from "./Icon";
+import {IconButtonProps} from "./Common";
+import {isMobileDevice} from "./MediaQuery";
 import {Modal} from "./Modal";
 import {Text} from "./Text";
+import {ThemeContext} from "./Theme";
 import {Tooltip} from "./Tooltip";
 import {Unifier} from "./Unifier";
+import {isNative} from "./Utilities";
 
-// eslint-disable-next-line react/display-name
-export const IconButton = forwardRef(
-  (
-    {
-      prefix,
-      icon,
-      onClick,
-      size,
-      type = "primary",
-      withConfirmation = false,
-      confirmationText = "Are you sure you want to continue?",
-      confirmationHeading = "Confirm",
-      tooltip,
-      indicator,
-      indicatorNumber,
-      indicatorStyle = {position: "bottomRight", color: "primary"},
-      testID,
-    }: IconButtonProps,
-    ref
-  ) => {
-    const [showConfirmation, setShowConfirmation] = useState(false);
+const ConfirmationModal: FC<{
+  visible: boolean;
+  title: string;
+  subTitle?: string;
+  text: string;
+  onConfirm: () => void;
+  onCancel: () => void;
+}> = ({visible, title, subTitle, text, onConfirm, onCancel}) => {
+  return (
+    <Modal
+      primaryButtonOnClick={onConfirm}
+      primaryButtonText="Confirm"
+      secondaryButtonOnClick={onCancel}
+      secondaryButtonText="Cancel"
+      subTitle={subTitle}
+      title={title}
+      visible={visible}
+      onDismiss={onCancel}
+    >
+      <Text>{text}</Text>
+    </Modal>
+  );
+};
 
-    // TODO: support hover states for icon button.
+const IconButtonComponent: FC<IconButtonProps> = ({
+  accessibilityLabel,
+  confirmationHeading = "Confirm",
+  confirmationText = "Are you sure you want to continue?",
+  iconName,
+  loading: propsLoading = false,
+  testID,
+  variant = "primary",
+  withConfirmation = false,
+  onClick,
+}) => {
+  const [loading, setLoading] = useState(propsLoading);
+  const [showConfirmation, setShowConfirmation] = useState(false);
+  const {theme} = useContext(ThemeContext);
 
-    const opacity = 1;
-    let bgColor: SurfaceColor;
-    let iconColor: IconColor;
-    // TODO match up icon colors better
-    if (type === "muted") {
-      bgColor = "base";
-      iconColor = "primary";
-    } else if (type === "secondary") {
-      bgColor = "neutralLight";
-      iconColor = "secondaryDark";
-    } else {
-      bgColor = "primary";
-      iconColor = "inverted";
-    }
+  if (!theme) {
+    return null;
+  }
```

Comment:
If theme is unaccessible it will cause the app to crash. We could add a loading button, however i think we'll have theme called in init app opening with one loader as opposed to a loader image on every component

Reply:
Have you encountered that at all? Should we be also returning null if no theme in all the other components, then?